thanos/charts/alerts/alerts.yaml

```diff
@@ -0,0 +1,668 @@
+        DNS queries for store endpoints.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryhighdnsfailures
+      summary: Thanos Query is having high number of DNS failures.
+    expr: |
+      (
+        sum by (job) (rate(thanos_query_store_apis_dns_failures_total{job=~".*thanos-query.*"}[5m]))
+      /
+        sum by (job) (rate(thanos_query_store_apis_dns_lookups_total{job=~".*thanos-query.*"}[5m]))
+      ) * 100 > 1
+    for: 15m
+    labels:
+      severity: warning
+  - alert: ThanosQueryInstantLatencyHigh
+    annotations:
+      description: Thanos Query {{$labels.job}} has a 99th percentile latency of {{$value}}
+        seconds for instant queries.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryinstantlatencyhigh
+      summary: Thanos Query has high latency for queries.
+    expr: |
+      (
+        histogram_quantile(0.99, sum by (job, le) (rate(http_request_duration_seconds_bucket{job=~".*thanos-query.*", handler="query"}[5m]))) > 40
+      and
+        sum by (job) (rate(http_request_duration_seconds_bucket{job=~".*thanos-query.*", handler="query"}[5m])) > 0
+      )
+    for: 10m
+    labels:
+      severity: critical
+  - alert: ThanosQueryRangeLatencyHigh
+    annotations:
+      description: Thanos Query {{$labels.job}} has a 99th percentile latency of {{$value}}
+        seconds for range queries.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryrangelatencyhigh
+      summary: Thanos Query has high latency for queries.
+    expr: |
+      (
+        histogram_quantile(0.99, sum by (job, le) (rate(http_request_duration_seconds_bucket{job=~".*thanos-query.*", handler="query_range"}[5m]))) > 90
+      and
+        sum by (job) (rate(http_request_duration_seconds_count{job=~".*thanos-query.*", handler="query_range"}[5m])) > 0
+      )
+    for: 10m
+    labels:
+      severity: critical
+  - alert: ThanosQueryOverload
+    annotations:
+      description: Thanos Query {{$labels.job}} has been overloaded for more than
+        15 minutes. This may be a symptom of excessive simultaneous complex requests,
+        low performance of the Prometheus API, or failures within these components.
+        Assess the health of the Thanos query instances, the connected Prometheus
+        instances, look for potential senders of these requests and then contact support.
+      runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosqueryoverload
+      summary: Thanos query reaches its maximum capacity serving concurrent requests.
+    expr: |
+      (
+        max_over_time(thanos_query_concurrent_gate_queries_max[5m]) - avg_over_time(thanos_query_concurrent_gate_queries_in_flight[5m]) < 1
+      )
+    for: 15m
+    labels:
+      severity: warning
+- name: thanos-receive
```

Comment:
Do we always want to deploy all alerts for all Thanos components? Or enable/disable them via values?

Reply:
trying out a way to handle the enabling of the alerts